susemanager-utils/susemanager-sls/src/modules/mgractionchains.py

```diff
@@ -57,22 +57,39 @@ def _get_ac_storage_filenamepath():
 
     return os.path.join(minion_d_dir, '_mgractionchains.conf')
 
+def check_reboot_required(target_sls):
+    sls_file_on_minion = __salt__['cp.cache_file']('{0}{1}.sls'.format('salt://actionchains/', target_sls.replace('actionchains.','')))
+    yaml_file = _read_sls_file(sls_file_on_minion)
+    list_param = yaml_file['schedule_next_chunk']['mgrcompat.module_run']
+
+    for dic in list_param:
+        if 'reboot_required' in dic.keys() :
+            return dic.values()
+    return False
+
 def _read_next_ac_chunk(clear=True):
     '''
     Read and remove the content of '_mgractionchains.conf' file. Return the parsed YAML.
     '''
     f_storage_filename = _get_ac_storage_filenamepath()
     if not os.path.isfile(f_storage_filename):
         return None
+    ret = _read_sls_file(f_storage_filename)
+    if clear:
+        os.remove(f_storage_filename)
+    return ret
+
+def _read_sls_file(filename):
+    if not os.path.isfile(filename):
+        log.error("File {0} not found".format(filename))
+        return None
     ret = None
     try:
-        with fopen(f_storage_filename, "r") as f_storage:
-            ret = yaml.load(f_storage.read())
-        if clear:
-            os.remove(f_storage_filename)
+        with fopen(filename, "r") as f:
```

Comment:
Try to use `salt.utils.yamlloader.load` instead. Not sure if it's really safe here.